tycho-core/src/main/java/org/eclipse/tycho/core/osgitools/OsgiBundleProject.java

```diff
@@ -242,6 +247,11 @@ public void resolveClassPath(MavenSession session, MavenProject project) {
         addPDESourceRoots(project);
     }
 
+    private Collection<ClasspathEntry> computeExtraTestClasspath(ReactorProject reactorProject) {
+        // TODO read .classpath and so on to resolve to extra test classpath entries
```

Comment:
Actually that would be the interesting part for me, how do I hook into this without 

a) coding all possible classpath container right here (a container might be test related or not by configuration)
b) how could i contribute to the resolving phases so the required deps are actually there?

Reply:
a. What existing containers do you have in mind in practice? I think in any case, there needs to be code that "expands" a containers to extra requirements/classpath entries somewhere, and Tycho could support the typical ones.

b. See later: this steps happens *after* the p2&OSGi resolution, which do take care of resolving deps.

Reply:
But how P2&OSGi knows that there are required? Would that not mean we need to put classpath parsing here and there?

For the containers, I think tycho should support JUNIT, UserLib  and maybe JRE as an alternative to BREE (but thats more for the future) ClasspathVariables could also be useful on the long term, but as containers are extensible one would need some way to provide a tycho(-extras) plugin to support specific ones. For example I'm currently develop a plugin that will contain other classpath types so I'd like to supply users with a small plugin to support this special one too.

In essence it should be possible to put 
https://github.com/eclipse/tycho/blob/400c244e564cf2f52744a032f284e271ca43ffe5/tycho-surefire/tycho-surefire-plugin/src/main/java/org/eclipse/tycho/surefire/container/JunitClassPathContainer.java
into an own plugin, it adds the necessary requirements and everything else gets aware of them.

Reply:
> Would that not mean we need to put classpath parsing here and there?

Yes, here and there exactly, as suggested by the TODO

> For the containers, I think tycho should support JUNIT, UserLib

OK, so let's just fill this TODO.

> ClasspathVariables could also be useful on the long term

OK, but let's make it another iteration later.

>  For example I'm currently develop a plugin that will contain other classpath types so I'd like to supply users with a small plugin to support this special one too.

That will require people who use your container and need it at build time to also add an extra tycho extension for there build to resolve the classpath container to a set of dependencies/classpath entries.
So basically enforcing new extensibility and so on. Why not, but better to postpone it in a further step, when we have support for standard containers working well enough to get clear ideas of how to factorize better.

> into an own plugin, it adds the necessary requirements and everything else gets aware of them.

It's too early to go that way IMO. We first need to have support for standard things (JUnit) working decently before making this extensible.

Reply:
I don't think so. If JUnit would work generic others would work also. That's the whole purpose of #60 to **not** add special handling on different places but allow **one** point of extension.

Reply:
Sorry, but I really don't want to start by adding extensibility at the same time as we're adding other complexity on the lifecycle + compiler/surefire mojos. Just take smaller steps, with more focused changes each providing a valuable piece of the puzzle which has functional value on its own, and everything will work better for everyone.

Reply:
> OK, so let's just fill this TODO.

So what's next, do you mean I should try to add something at the "todo points" or should the PR be merged and we later fill the TODOs?